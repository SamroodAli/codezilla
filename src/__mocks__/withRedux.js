import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import PropTypes from 'prop-types';
import data from '../../public/data/courses.json';

const initialState = {
  videos: [data],
  filters: {
    React: true,
    Angular: true,
    Vue: true,
    'Functional Programming': true,
    'Object Oriented Programming': true,
    Microservices: true,
    'JAM Stack': true,
  },
  theme: 'light',
  currentVideo: null,
};

const mockStore = configureStore([reduxThunk]);
export const store = mockStore(initialState);

const WithRedux = ({ children }) => (
  <Provider store={store}><StaticRouter>{children}</StaticRouter></Provider>
);

WithRedux.propTypes = {
  children: PropTypes.node.isRequired,
};

export default WithRedux;
