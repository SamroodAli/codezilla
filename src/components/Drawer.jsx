import { useState } from 'react';
import { Drawer, ANCHOR } from 'baseui/drawer';
import { Button } from 'baseui/button';
import Filters from './Filter';

const FilterOptions = () => {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <>
      <Button onClick={() => setIsOpen(true)}>Filter Courses</Button>
      <Drawer
        isOpen={isOpen}
        autoFocus
        onClose={() => setIsOpen(false)}
        anchor={ANCHOR.bottom}
      >
        <Filters />
      </Drawer>
    </>
  );
};

export default FilterOptions;
