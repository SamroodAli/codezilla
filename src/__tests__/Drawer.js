import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import withRedux from 'withRedux';
import Drawer from '../components/Drawer';

describe('Testing Drawer with redux', () => {
  it('Drawer is defined', async () => {
    const courses = render(withRedux(<Drawer />));
    expect(courses).toBeDefined();
  });

  it('Drawer snapshot testing', () => {
    const tree = renderer.create(withRedux(<Drawer />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
