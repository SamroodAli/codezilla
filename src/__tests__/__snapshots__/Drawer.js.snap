// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Drawer with redux Drawer snapshot testing 1`] = `
<button
  className=""
  data-baseweb="button"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
>
  Filter Courses
</button>
`;
